internetarchive/openlibrary
# Add star ratings to "already read" bookshelf items

Closes #1032


Adds star ratings to each list item on the owner's `/account/books/already-read` page.

### Technical

Most of the changes were made to StarRatings.html:
- `Edition` was made optional. Edition is not readily available in the reading log template (as far as I know), and star ratings are captured at the work level.
- Redirect path `/account/books/already-read` is passed to the template when rendered on the reading log page.  This prevents the default redirect to the work's page.
- Form elements can now be uniquely identified, ensuring that multiple star ratings components work properly when they appear on a single page.
- The modal link can be hidden with the `show_modal_link` parameter.

### Testing

1. While logged in, visit `/account/books/already-read`.
2. Ensure that star rating components are present, but modal links are not.
3. Change the rating of the first list item. Ensure that the page reloads and the new rating is displayed.
4. Change the rating of another list item. Ensure that that only list item's rating has changed.
5. Delete a rating. Ensure that the list item has no stars when the page reloads.
6. View this page in a private browser, ensuring that no star ratings are present.
7. While logged in, navigate to a book page.
8. Ensure that the star rating component is in the list widget, and that the modal link is present.
9. Change the star rating on the work page. Ensure that the page reloads and that your new rating is displayed in the component.

### Screenshot

![Screenshot from 2021-11-23 19-31-05](https://user-images.githubusercontent.com/28732543/143159901-47eb54f1-0e3b-43e2-bc14-e13d3c473653.png)

![Screenshot from 2021-11-23 19-32-12](https://user-images.githubusercontent.com/28732543/143159933-d47d6a92-4b29-4530-9aa1-f037ba66813a.png)

### Stakeholders
